### PR TITLE
httpie: 2.2.0 -> 2.4.0

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Motivation for this change
Replaces #102466

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112381` at 56b97d57 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/d9de96e4ffd6197de575d0c5b2bf2eee">http-prompt</a></li></li>
    <li><a href="https://gist.github.com/8e1d550be643ed8258b49e10770a5f7e">httpie</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/tools/networking/httpie/default.nix:15:5:
```
   |
15 |     ./strip-venv.patch
   |     ^
```
</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112381` at 56b97d57 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/92d2fee99c3860a3868a226c3869bc10">http-prompt</a></li></li>
    <li><a href="https://gist.github.com/d62f82d136d8ba221dbc8ce50a6b6ce6">httpie</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/tools/networking/httpie/default.nix:15:5:
```
   |
15 |     ./strip-venv.patch
   |     ^
```
</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 112381` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>http-prompt</li>
    <li>httpie</li>
  </ul>
</details>

File: pkgs/tools/networking/httpie/strip-venv.patch [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
  # HACK: hardcoded paths, venv should be irrelevant, etc.
- # TODO: replaces the process with Python code
+ # TODO: simplify by using the Python API instead of a subprocess
+ #       then we wont’t need the paths.
 -VENV_BIN = Path(__file__).parent.parent / 'venv/bin'
 -VENV_PYTHON = VENV_BIN / 'python'
 -VENV_RST2PSEUDOXML = VENV_BIN / 'rst2pseudoxml.py'
 +VENV_PYTHON = 'python'
-+VENV_RST2PSEUDOXML = 'rst2pseudoxml.py'
++VENV_RST2PSEUDOXML = 'rst2pseudoxml.p'
```
Comment (fabaff) [inline pkgs/tools/networking/httpie/strip-venv.patch:13]: Typo?

```text
'rst2pseudoxml.p'
```

or not as there are no errors in the log.

**Review by fabaff — COMMENTED** [review]
- [x] package build (see https://github.com/NixOS/nixpkgs/pull/112381#issuecomment-775181375)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-112381]$ ./results/httpie/bin/http https://reqbin.com/echo/get/json
HTTP/1.1 200 OK
Accept-Ranges: bytes
CF-Cache-Status: HIT
CF-RAY: 61e6257d4aca375c-MXP
Cache-Control: max-age=31536000
Connection: keep-alive
Content-Length: 19
Content-Type: application/json
Date: Mon, 08 Feb 2021 14:47:32 GMT
Expect-CT: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
NEL: {"report_to":"cf-nel","max_age":604800}
Report-To: {"max_age":604800,"group":"cf-nel","endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report?s=l%2FG68d8ocqngDlG5Mm4c%2Fm41yjAGmuKkJzWFUb0A%2BOLFd0y4NuqdD3XFCmJvA2BrEPKylP%2B88ravA6RmbaynQQdw289xeTZP0ul6"}]}
Server: cloudflare
Set-Cookie: __cfduid=d0d99c5ddf32e09e4e8b0c1acf3452f971612795652; expires=Wed, 10-Mar-21 14:47:32 GMT; path=/; domain=.reqbin.com; HttpOnly; SameSite=Lax; Secure
Vary: Accept-Encoding
cf-request-id: 0823b5c2490000375cffbcc000000001

{
    "success": "true"
}
[...]
```

Looks good to me functionwise.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112381` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>http-prompt</li>
<li>httpie</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112381` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>http-prompt</li>
<li>httpie</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112381` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>http-prompt</li>
<li>httpie</li>
</ul>
</details>

**Pull request merged**